packages/sensors_plus/tizen/src/sensors_plus_plugin.cc

```diff
@@ -67,40 +71,87 @@ class SensorsPlusPlugin : public flutter::Plugin {
+        });
+
     std::unique_ptr<FlEventChannel> accelerometer_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/accelerometer",
             &flutter::StandardMethodCodec::GetInstance());
     accelerometer_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(
-            SensorType::kAccelerometer));
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kAccelerometer,
+                                                    interval_));
 
     std::unique_ptr<FlEventChannel> gyroscope_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/gyroscope",
             &flutter::StandardMethodCodec::GetInstance());
     gyroscope_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kGyroscope));
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kGyroscope,
+                                                    interval_));
 
     std::unique_ptr<FlEventChannel> user_accel_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/user_accel",
             &flutter::StandardMethodCodec::GetInstance());
     user_accel_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kUserAccel));
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kUserAccel,
+                                                    interval_));
 
     std::unique_ptr<FlEventChannel> magnetometer_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/magnetometer",
             &flutter::StandardMethodCodec::GetInstance());
     magnetometer_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kMagnetometer));
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kMagnetometer,
+                                                    interval_));
   }
+
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    const std::string &method_name = method_call.method_name();
+    const flutter::EncodableValue *arguments = method_call.arguments();
+
+    if (method_name == "setAccelerationSamplingPeriod" ||
+        method_name == "setGyroscopeSamplingPeriod" ||
+        method_name == "setUserAccelerometerSamplingPeriod" ||
+        method_name == "setMagnetometerSamplingPeriod") {
+      const auto *sampling_period = std::get_if<int32_t>(arguments);
+      if (sampling_period) {
+        // TODO(jsuya): In the sensor_plus_platform_interface, the
+        // set***SamplingPeriod method call is called only before creating an
```

Comment:
Why don't you unset the event callback and set the callback again?

Reply:
I didn't check the example code before writing these comments.
The example code calls each `set*SamplingPeriod` to be set separately from the event stream creation. So I added `SetInterval()`, which calls `sensor_listener_set_interval()`.